pyi.py

```diff
@@ -1460,10 +1432,9 @@ def _check_subscript_version_check(self, node: ast.Compare) -> None:
         version_info = node.left
         if isinstance(version_info, ast.Subscript):
             slc = version_info.slice
-            # TODO: ast.Num works, but is deprecated
-            if isinstance(slc, ast.Num):
+            if isinstance(slc, ast.Constant):
                 # anything other than the integer 0 doesn't make much sense
-                if isinstance(slc.n, int) and slc.n == 0:
+                if isinstance(slc.value, int) and slc.value == 0:
```

Comment:
Before we kept doing `type(slc.value) is int`, is there any reason for that? In general we should prefer `isinstance()`.

Reply:
I obviously prefer `isinstance()` checks for Python code in general. But for ASTs specifically, I think there are often reasons to prefer doing exact identity checks. Something pretty weird is going on if we somehow get an `ast.Constant` node where the value is an instance of a subclass of `int`. I don't know how that would happen or how we'd test for it.

Having said that, I don't really have a specific reason to avoid using `isinstance()` in any of these places, so we can probably switch to using `isinstance()` if you like!

Reply:
Here specifically, we were already using `isinstance()`, and I didn't bother to change it as it seemed orthogonal to the purpose of this PR

Reply:
Yes, just reading over this PR, we're using type() maybe 75% of the time for no clear reason. But it doesn't really cause any concrete problems, so no need to fix it now.

Reply:
It is probably because `isinstance(True, int)`.

Reply:
Oh good point, I forgot about that.